api/crossmodelrelations/crossmodelrelations.go

```diff
@@ -28,85 +32,209 @@ func (c *Client) Close() error {
 	return c.ClientFacade.Close()
 }
 
+// handleError is used to process an error obtained when making a facade call.
+// If the error indicates that a macaroon discharge is required, this is done
+// and the resulting discharge macaroons passed back so the api call can be retried.
+func (c *Client) handleError(err error) (macaroon.Slice, error) {
+	if params.ErrCode(err) != params.CodeDischargeRequired {
+		return nil, err
```

Comment:
Not needed here IMO because the error is wrapped by the facade methods which call this handleError().
Tracing doesn't really do much for us here.